### PR TITLE
[build-script] Ensure source directory argument to be the last of cmake invocation

### PR DESCRIPTION
**Pull request by rintaro**
**Warning:** This PR conflicts with #2283. I will rebase this PR if it's merged.
#### What's in this pull request?

Although `cmake -DFOO=BAR /path/to/source -DBAZ=QUX` works,
[man page of `cmake(1)`](https://cmake.org/cmake/help/v3.2/manual/cmake.1.html) states `<path-to-source>` is the last argument.

> `cmake [<options>] (<path-to-source> | <path-to-existing-build>)`

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

### CODE REVIEW

File: utils/build-script-impl [hunk 1]
```diff
@@ -1552,6 +1552,7 @@ for deployment_target in "${HOST_TARGET}" "${CROSS_TOOLS_DEPLOYMENT_TARGETS[@]}"
 
     for product in "${PRODUCTS[@]}"; do
         unset skip_build
+        source_dir=$(eval echo \${$(toupper ${product})_SOURCE_DIR})
```
Comment (gribozavr) [inline utils/build-script-impl]: Could we remove the `eval`?

```shell
X_SD=1
Y_SD=2
product=x
source_dir_var="$(toupper ${product})_SD"
echo "${!source_dir_var}" # 1
product=y
source_dir_var="$(toupper ${product})_SD"
echo "${!source_dir_var}" # 2
```

**Comment by gribozavr** [comment]
LGTM modulo the comment.

**Comment by gribozavr** [comment]
@swift-ci Please test

File: utils/build-script-impl (hunk 1 above)
Comment (rintaro) [inline utils/build-script-impl]: Will do that.

File: utils/build-script-impl [hunk 2]
```diff
@@ -1552,6 +1552,8 @@ for deployment_target in "${HOST_TARGET}" "${CROSS_TOOLS_DEPLOYMENT_TARGETS[@]}"
 
     for product in "${PRODUCTS[@]}"; do
         unset skip_build
+        source_dir_var="$(toupper ${product})_SOURCE_DIR"
+        source_dir=${!source_dir_var}
```
Comment (rintaro) [inline utils/build-script-impl]: OK now?

Comment (gribozavr) [inline utils/build-script-impl]: Thank you!

**Comment by gribozavr** [comment]
@swift-ci Please test and merge

**Pull request merged**